src/script/components/InputBar/InputBar.tsx

```diff
@@ -146,83 +138,75 @@ const InputBar = ({
 
+  // Right sidebar
   const {rightSidebar} = useAppMainState.getState();
   const lastItem = rightSidebar.history.length - 1;
   const currentState = rightSidebar.history[lastItem];
   const isRightSidebarOpen = !!currentState;
 
   const inputPlaceholder = messageTimer ? t('tooltipConversationEphemeral') : t('tooltipConversationInputPlaceholder');
 
-  const candidates = participatingUserEts.filter(userEntity => !userEntity.isService);
-  const mentionSuggestions = editedMention ? searchRepository.searchUserInSet(editedMention.term, candidates) : [];
-
-  const isEditing = !!editMessageEntity;
+  const isEditing = !!editedMessage;
   const isReplying = !!replyMessageEntity;
   const isConnectionRequest = isOutgoingRequest || isIncomingRequest;
-  const hasLocalEphemeralTimer = isSelfDeletingMessagesEnabled && localMessageTimer && !hasGlobalMessageTimer;
-
-  const richTextInput = getRichTextInput(currentMentions, inputValue);
+  const hasLocalEphemeralTimer = isSelfDeletingMessagesEnabled && !!localMessageTimer && !hasGlobalMessageTimer;
 
   // To be changed when design chooses a breakpoint, the conditional can be integrated to the ui-kit directly
   const isScaledDown = useMatchMedia('max-width: 768px');
 
-  const config = {
-    GIPHY_TEXT_LENGTH: 256,
-  };
+  const showGiphyButton = messageContent.text.length > 0 && messageContent.text.length <= config.GIPHY_TEXT_LENGTH;
 
-  const showGiphyButton = inputValue.length > 0 && inputValue.length <= config.GIPHY_TEXT_LENGTH;
-
-  const updateSelectionState = (updateOnInit = true) => {
-    if (!updateOnInit) {
-      return;
-    }
-
-    if (!textareaRef.current) {
-      return;
-    }
-
-    const {selectionStart: start, selectionEnd: end} = textareaRef.current;
-    const {newEnd, newStart} = getSelectionPosition(textareaRef.current, currentMentions);
-
-    if (newStart !== start || newEnd !== end) {
-      textareaRef.current.selectionStart = newStart;
-      textareaRef.current.selectionEnd = newEnd;
-    }
+  const shouldReplaceEmoji = useUserPropertyValue(
+    () => propertiesRepository.getPreference(PROPERTIES_TYPE.EMOJI.REPLACE_INLINE),
+    WebAppEvents.PROPERTIES.UPDATE.EMOJI.REPLACE_INLINE,
+  );
 
-    setSelectionStart(newStart);
-    setSelectionEnd(newEnd);
+  // Mentions
+  const getMentionCandidates = (search?: string | null) => {
+    const candidates = conversation.participating_user_ets().filter(userEntity => !userEntity.isService);
+    return typeof search === 'string' ? searchRepository.searchUserInSet(search, candidates) : candidates;
   };
 
-  const moveCursorToEnd = (endPosition: number, updateSelection = true) => {
-    updateSelectionState(updateSelection);
-    setTimeout(() => {
-      textareaRef.current?.setSelectionRange(endPosition, endPosition);
-      textareaRef.current?.focus();
-    }, 0);
-  };
+  useTypingIndicator({
+    isEnabled: isTypingIndicatorEnabled,
+    text: messageContent.text,
+    onTypingChange: useCallback(
```

Comment:
![Screenshot 2023-09-05 at 10 00 13](https://github.com/wireapp/wire-webapp/assets/1090716/9ebab627-2fb1-404f-a45c-8e163fdc618e)

It's actually used in an effect in the hook. So we want to be careful about re-creating this function at each render (it would cause the effect to trigger on each render)

Reply:
Yeah, `useCallback` make sense in general, but what I mean is: shouldn't we store a function that is returned by useCallback in some variable in component's body? Not sure if the definition is still cached if we use it this way 🤔

Reply:
Oh, it is. That the purpose of `useCallback` it's going to return the same reference if the deps array doesn't change. 
It doesn't matter if you store it in a variable or not, it's the `useCallback` function that works like this

Reply:
Okay makes sense, all good then 😄